### PR TITLE
Linked Images endpoint to the Model

### LINKED ISSUE #15: Link the calling of the logo detection model(s) to the endpoint
In this we want to build out the `ImageProcessingService` and `LogoDetectionService` in our `images` app.

Suggested flow:
* At the `POST /images/predictions/` endpoint, pass the images to the `ImageProcessingService().process_images(...)` method (the names are not strict just so long as they make sense)
* For each image we want to do something like calling a `detect_logos(...)` method on the `LogoDetectionService` class.
* We will need to save the images locally here (I would recommend doing it in the `ImageProcessingService`). #53 will address this so that the images are saved locally and discarded afterwards
* The model team will have created a `detect_logos` or similar function in `ml/models/logo_detection`, we want to import this and pass the **path** of the image we have saved into it
* That method will give the bounding box coordinates and a confidence interval, store these and return them

Suggested response from the API:
```json
{
    "image0": [{
        "bbox": {
            "x1": 100,
            "x2": 34.5,
            "y1": 6.7,
            "y2": 12.4,
        },
        "confidence": 76.5,
    }],
    "my_image": [{
        "bbox": {
            "x1": 100,
            "x2": 34.5,
            "y1": 6.7,
            "y2": 12.4,
        },
        "confidence": 76.5,
    }, {
        "bbox": {
            "x1": 12,
            "x2": 34.5,
            "y1": 6.7,
            "y2": 12.4,
        },
        "confidence": 36.5,
    }],
    "another_image": []
}
```
The outermost keys (`"image0", "my_image", "another_image"`) should be the keys of the request body (i.e. they are sent by the client).

Tips for accessing the files from the request body:
* The payload of a request is available from `request.data` which returns a dictionary that will look like
```python
request.data = {
    "my_image": "<my_image_blob>",
    "another_image": "<another_image_blob>",
    "image0": "<image0_blob>"
}
```
* The type of the values of the dictionary will be `django.core.files.uploadedfile.InMemoryUploadedFile`

Considerations:
* We don't care about the order of the response, either with respect to the keys or the values of the list, just so long as the bboxes and confidence intervals correspond to each other and the image they are assigned
* If we error on an image, we want to continue with any remaining images, that is the processing of each image of the payload should not depend on any processing of the other images

### LINKED ISSUE #28: Add endpoints that allow the user to upload an image and feed it through the image classification models

### PR DESCRIPTION
**Pull request by dhrvrc**

### CODE REVIEW

File: api/images/services/LogoDetectionService.py [hunk 1]
```diff
@@ -1,2 +1,19 @@
+from ml.models.logo_detection.process import process as detectLogos
```
Comment (sktylr) [inline api/images/services/LogoDetectionService.py]: `snake_case`
```suggestion
from ml.models.logo_detection.process import process as detect_logos
```

File: api/images/services/LogoDetectionService.py [hunk 2]
```diff
@@ -1,2 +1,19 @@
...
+
+# from rest_framework import serializers
+
+
 class LogoDetectionService:
-    pass
+    def detect_in_image(self, imagePath):
```
Comment (sktylr) [inline api/images/services/LogoDetectionService.py]: `snake_case`
```suggestion
    def detect_in_image(self, image_path):
```

File: api/images/views.py [hunk 3]
```diff
@@ -1,16 +1,61 @@
...
+from rest_framework import generics, status
 from rest_framework.response import Response
 
+from .services.ImageProcessingService import ImageProcessingService
+
 
 # Create your views here.
 class PredictionsViewSet(generics.GenericAPIView):
     def post(self, request):
+        print(request.data)
```
Comment (sktylr) [inline api/images/views.py]: rm

File: api/images/services/ImageProcessingService.py
Comment (sktylr) [inline api/images/services/ImageProcessingService.py:1]: Let's change this file name to `image_processing_service.py` to conform to `snake_case` conventions. You can use `git mv` to rename whilst preserving the git history

File: api/images/services/LogoDetectionService.py
Comment (sktylr) [inline api/images/services/LogoDetectionService.py:1]: Likewise let's change to `logo_detection_service.py`

**Review by sktylr — CHANGES_REQUESTED** [review]
Nice work! It is working well locally and great to get this done in this sprint

File: api/images/services/logo_detection_service.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from ml.models.logo_detection.process import process as detect_logos
+
+# from rest_framework import serializers
+
+
+class logo_detection_service:
```
Comment (sktylr) [inline api/images/services/logo_detection_service.py]: Sorry probably wasn't clear enough lol, class names should be `UpperCamelCase` and then file names, variable names and method / function names should be `snake_case`
```suggestion
class LogoDetectionService:
```

File: api/images/services/image_processing_service.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
+from api.utils.make_temp_directory import make_temp_directory
+
+from .logo_detection_service import logo_detection_service
+
+
+class image_processing_service:
```
Comment (sktylr) [inline api/images/services/image_processing_service.py]:
```suggestion
class ImageProcessingService:
```

**Review by egaynor — APPROVED** [review]

**Review by sktylr — APPROVED** [review]
🏆

**Pull request merged**